code/model/modelinterp.cpp

```diff
@@ -1689,28 +1691,41 @@ int check_values(vec3d *N)
 	return 0;
 }
 
+// Unpacks verts from TMAPPOLY since it's no longer direct memory mapped with model_tmap_vert
+void unpack_tmap_verts(const ubyte* vs, model_tmap_vert* verts, uint n_vert) {
+	// Copy the verts manually since they aren't aligned with the struct
+	for (uint i = 0; i < n_vert; i++) {
+		verts[i].vertnum = us(vs);
+		verts[i].normnum = us(vs + 2);
+		verts[i].u = fl(vs + 4);
+		verts[i].v = fl(vs + 8);
+		vs += 12;
+	}
+}
+
 int Parse_normal_problem_count = 0;
 
 void parse_tmap(int offset, ubyte *bsp_data)
 {
 	int pof_tex = w(bsp_data+offset+40);
-	int n_vert = w(bsp_data+offset+36);
+	uint n_vert = uw(bsp_data+offset+36);
 	
 	ubyte *p = &bsp_data[offset+8];
-	model_tmap_vert *tverts;
+	auto tverts = new model_tmap_vert[n_vert];
 
-	tverts = (model_tmap_vert *)&bsp_data[offset+44];
+	// Copy the verts manually since they aren't aligned with the struct
+	unpack_tmap_verts(&bsp_data[offset + 44], tverts, n_vert);
```

Comment:
caution, is accessing the 44th element instead of 44th byte.

Reply:
`bsp_data` is an array of bytes, so it is both.

Reply:
yeah, but it uses a different syntax instead of the "usual" pointer arithmetic most other changes do. Hence the Caution